cloudinit/sources/helpers/vmware/imc/config_nic.py

```diff
@@ -87,161 +80,190 @@ def find_devices(self):
             name = section.split(":", 1)[0]
             self.mac2Name[mac] = name
 
-    def gen_one_nic(self, nic):
+    def gen_one_nic_v2(self, nic):
         """
-        Return the config list needed to configure a nic
-        @return (list): the subnets and routes list to configure the nic
+        Return the config dict needed to configure a nic
+        @return (dict): the config dict to configure the nic
         @param nic (NicBase): the nic to configure
         """
         mac = nic.mac.lower()
         name = self.mac2Name.get(mac)
         if not name:
             raise ValueError("No known device has MACADDR: %s" % nic.mac)
 
-        nics_cfg_list = []
-
-        cfg = {"type": "physical", "name": name, "mac_address": mac}
-
-        subnet_list = []
-        route_list = []
-
-        # Customize IPv4
-        (subnets, routes) = self.gen_ipv4(name, nic)
-        subnet_list.extend(subnets)
-        route_list.extend(routes)
-
-        # Customize IPv6
-        (subnets, routes) = self.gen_ipv6(name, nic)
-        subnet_list.extend(subnets)
-        route_list.extend(routes)
-
-        cfg.update({"subnets": subnet_list})
-
-        nics_cfg_list.append(cfg)
-        if route_list:
-            nics_cfg_list.extend(route_list)
-
-        return nics_cfg_list
-
-    def gen_ipv4(self, name, nic):
-        """
-        Return the set of subnets and routes needed to configure the
-        IPv4 settings of a nic
-        @return (set): the set of subnet and routes to configure the gateways
-        @param name (str): subnet and route list for the nic
-        @param nic (NicBase): the nic to configure
-        """
-
-        subnet = {}
-        route_list = []
-
+        nic_config_dict = {}
+        # match
+        match = self.gen_match(mac)
+        if match:
```

Comment:
`match` will always be non-None so this check isn't necessary.

Reply:
Updated this part as @sshedi suggested, thanks.